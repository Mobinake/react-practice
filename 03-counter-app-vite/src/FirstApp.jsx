import PropTypes from 'prop-types';

export const FirstApp = ({ title, subTitle }) => {
	return (
		<>
			<h1 data-testid='test-title'>{title}</h1>
			<p>{subTitle}</p>
			<p>Prueba de comp</p>
		</>
	);
};

FirstApp.propTypes = {
	title: PropTypes.string.isRequired,
	subTitle: PropTypes.number.isRequired,
};

FirstApp.defaultProps = {
	title: 'No hay t√≠tulo',
	subTitle: 100,
};
