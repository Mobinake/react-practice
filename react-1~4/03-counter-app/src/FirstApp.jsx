import PropTypes from 'prop-types';

const newMessage = {
	message: 'Hola, react.',
	title: 'Mobin',
};

export function FirstApp({ title, subtitle }) {
	return (
		<>
			{/* se puede colocar cualquier codigo de JS aqui, excepto funciones.
                aqui lo que hace es convertir la respuesta en JSON, para que no se devuelva como funcion*/}
			<code>{JSON.stringify(newMessage)}</code>
			<h1 data-testid='test-title'>{title}</h1>
			<h2>{subtitle}</h2>
		</>
	);
}

FirstApp.propTypes = {
	title: PropTypes.string.isRequired,
	subtitle: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
	title: 'No hay titulo',
	subtitle: 'No hay subtitulo',
};
