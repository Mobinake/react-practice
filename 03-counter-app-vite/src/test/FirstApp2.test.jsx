/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react';
import { FirstApp } from '../FirstApp';

describe('Pruebas en <FirstApp2/>', () => {
	const title = 'Hola, soy Kakaroto';
	const subTitle = 'Soy el apellido';
	test('debe de hacer match con el snapshot', () => {
		const { container } = render(<FirstApp title={title} />);
		expect(container).toMatchSnapshot();
	});

	test('Debe mostrar el mensaje Hola, soy Kakaroto', () => {
		render(<FirstApp title={title} />);
		expect(screen.getByText(title)).toBeTruthy();
	});

	test('debe de mostrar el titulo en un h1', () => {
		render(<FirstApp title={title} />);
		expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(
			title
		);
	});

	test('debe mostrar el subtitulo enviado por props', () => {
		render(
			<FirstApp
				title={title}
				subTitle={subTitle}
			/>
		);
		expect(screen.getAllByText(subTitle).length).toBe(1);
	});
});
