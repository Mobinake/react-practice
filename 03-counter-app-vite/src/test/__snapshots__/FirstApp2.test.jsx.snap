// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp2/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Kakaroto
  </h1>
  <p />
  <p>
    Prueba de comp
  </p>
</div>
`;
